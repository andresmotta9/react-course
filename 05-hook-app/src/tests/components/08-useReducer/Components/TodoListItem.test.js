import React from 'react';
import { TodoListItem } from "../../../../components/08-useReducer/components/TodoListItem";
import { shallow } from 'enzyme'
import { demoTodos } from "../../../fixtures/demoTodos";

describe('Test on TodoListItem', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow( 
        <TodoListItem 
            todo={demoTodos[0]}
            index={0}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        />
    );

    test('should match <TodoListItem /> correctly', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('should call the handleDelete function', () => {

        wrapper.find('.btn').simulate('click');
        expect( handleDelete ).toHaveBeenCalled();
        expect( handleDelete ).toHaveBeenCalledWith( demoTodos[0].id );

    });

    test('should call the handleToggle function', () => {

        wrapper.find('p').simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith( demoTodos[0].id );

    });
    
    test('should the copy correctly', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(`${demoTodos[0].id} ${demoTodos[0].desc}`);

    });

    test('should has the complete class if TODO.done is true', () => {
        
        const todo = demoTodos[0];
        todo.done = true
        
        const wrapper = shallow( 
            <TodoListItem 
                todo={todo}
            />
        );

        expect( wrapper.find('p').hasClass('complete') ).toBe(true);

    });
    
})
