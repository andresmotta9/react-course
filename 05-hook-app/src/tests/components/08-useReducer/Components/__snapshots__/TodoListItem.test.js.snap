// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on TodoListItem should match <TodoListItem /> correctly 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
     
    Learn react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
